assets/js/controllers/export-history.controller.js

```diff
@@ -0,0 +1,88 @@
+angular
+  .module('walletApp')
+  .controller('ExportHistoryController', ExportHistoryController);
+
+function ExportHistoryController ($scope, $timeout, $translate, browser, format, DateHelper, Wallet, Alerts, ExportHistory, activeIndex) {
+  // Private
+  let compatibleBrowsers = browser.canDownloadFile;
+  let currentBrowser = browserDetection().browser;
+  let accounts = Wallet.accounts().filter(a => !a.archived && a.index !== null);
+  let addresses = Wallet.legacyAddresses().filter(a => !a.archived).map(a => a.address);
+  let all_accounts = {
+    index: '',
+    label: $translate.instant('ALL_ACCOUNTS'),
+    address: accounts.map(a => a.extendedPublicKey).concat(addresses)
+  };
+  let imported_addresses = {
+    index: 'imported',
+    label: $translate.instant('IMPORTED_ADDRESSES'),
+    address: addresses
+  };
+  let startDate = DateHelper.round(DateHelper.subtractDays(DateHelper.now(), 7));
+  let endDate = DateHelper.round(DateHelper.now());
+  let convertToDashed = (date) => { return DateHelper.toCustomShortDate('-', date); };
+  let convertToSlashed = (date) => { return DateHelper.toCustomShortDate('/', date); };
+
+  // Public
+  let vm = this;
+  vm.limit = 50;
+  vm.incLimit = () => vm.limit += 50;
+  vm.canTriggerDownload = compatibleBrowsers.indexOf(currentBrowser) > -1;
+  vm.format = DateHelper.format.shortDate;
+
+  vm.targets = [all_accounts].concat(accounts.map(format.origin));
+  if (addresses.length) {
+    vm.targets.push(imported_addresses);
+  }
+
+  vm.isLast = (t) => t === vm.targets[vm.limit - 1];
+
+  vm.activeCount = (
+    Wallet.accounts().filter(a => !a.archived).length +
+    Wallet.legacyAddresses().filter(a => !a.archived).length
+  );
+
+  vm.active = vm.activeCount === 1 ? all_accounts : vm.targets.filter(t => t.index.toString() === activeIndex)[0];
+
+  vm.options = {
+    minDate: DateHelper.bitcoinStartDate,
+    maxDate: DateHelper.now()
+  };
+
+  vm.start = {
+    open: true,
+    date: startDate
```

Comment:
Would have been nice to rename `startDate` to `initialStartDate`, but can do that some other time.